### PR TITLE
Add support for maturin backend

### PR DESCRIPTION
**Pull request by tetigi**
Adds support for [maturin](https://github.com/PyO3/maturin), ran via adding:

```
[build-system]
requires = ["maturin>=0.13,<0.14"]
build-backend = "maturin"
```

and adding `maturin` to the dev-dependencies.

This is invoked via `poetry`, so it's mostly just a rip of the `poetry` backend with the build invocation and destination swapped out.

### CODE REVIEW

**Review by wtfgecko — APPROVED** [review]

File: src/kraken/std/python/buildsystem/maturin.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    [build-system]
+    requires = ["maturin>=0.13,<0.14"]
+    build-backend = "maturin"
+    ```
+    """
+
+    name = "Maturin"
+
+    def __init__(self, project_directory: Path) -> None:
+        super().__init__(project_directory)
```
Comment (NiklasRosenstein) [inline src/kraken/std/python/buildsystem/maturin.py:33]: Don't need to override constructor if it only delegates to super

File: src/kraken/std/python/buildsystem/maturin.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            command = ["poetry", "version", as_version]
+            logger.info("%s", command)
+            sp.check_call(command, cwd=self.project_directory)
+
+        # We remove the contents of the dist folder to make sure we know what was produced.
+        wheels_dir = self.project_directory / "target" / "wheels"
+        if wheels_dir.exists():
+            shutil.rmtree(wheels_dir)
+
+        command = ["poetry", "run", "maturin", "build"]
```
Comment (NiklasRosenstein) [inline src/kraken/std/python/buildsystem/maturin.py:48]: Is this the only difference in the function? Maybe we should give the parent class a feature that allows overriding the command.

**Review by NiklasRosenstein — CHANGES_REQUESTED** [review]

File: src/kraken/std/python/buildsystem/maturin.py (hunk 2 above)
Comment (tetigi) [reply src/kraken/std/python/buildsystem/maturin.py:48]: It also affects the output directory (ie. rather than `dst_files` it's `wheels_dir`)

Comment (tetigi) [reply src/kraken/std/python/buildsystem/maturin.py]: Added specific methods which are overriden to provide those.

**Review by NiklasRosenstein — APPROVED** [review]

**Pull request merged**